interbotix_common_toolbox/interbotix_gravity_compensation/README.md

```diff
@@ -0,0 +1,227 @@
+```
+
+Step 3: the forces acting on the $i$'th link should balance. $\mathbf{f}_{i}\in\Bbb{S}$ is the force applied through the $i$'th joint to the $i+1$'th link and $\mathbf{f}^x_i\in\Bbb{S}$ is the net external force acting on the $i$'th link.
+
+```math
+\mathbf{f}^B_i=\mathbf{f}_{i-1}+\mathbf{f}^x_i-\mathbf{f}_{i}
+```
+
+The resulting torque of the $i$'th joint is given by
+
+```math
+\tau_i=\mathbf{s}_i\cdot\mathbf{f}_{i}
+```
+
+The inverse dynamics solver used in the package is ported from the [Orocos Kinematics and Dynamics Library (KDL)](https://www.orocos.org/kdl.html).
+Please refer to the KDL doc page for its [derivations](https://link.springer.com/book/10.1007/978-1-4899-7560-7) and [implementation](https://docs.ros.org/en/indigo/api/orocos_kdl/html/classKDL_1_1ChainIdSolver__RNE.html) details.
+
+## Motor Assistance
+
+The joint frictions cause two problems
+
+1. The kinetic friction resists the joint from moving.
+2. The disparity between the kinetic friction and the static friction causes [stiction](https://en.wikipedia.org/wiki/Stiction).
+
+Making things worse, the frictions at the joints are mapped to a resisting wrench at the end effector by the force Jacobian matrix. Depending on the joint positions of the arm, the resulting wrench may become infinitely large.
+
+We address this issue by using the motors to counteract the frictions proactively.
+
+### Kinetic Friction
+
+When a joint is moving, we add a assisting torque towards the direction of the motion.
+We use the coulomb friction model where the friction force is proportional to the total normal force acting on the contact surface.
+And we simplify the nominal wrench transmitted through the joint into two parts
+
+1. A torque about the joint axis.
+2. A constant wrench accounting for other forces and torques.
+
+As a result, the kinetic friction torque about the joint axis is given by
+
+```math
+\tau_k = \mu_k\tau+c
+```
+
+where $\tau_k$ is the kinetic friction torque, $\tau$ is the nominal torque transmitted through the joint, $\mu_k$ is the coefficient of kinetic friction associated with the torque (i.e., `kinetic_friction_coefficient`), $c$ accounts for the friction torque caused by other forces and torques (i.e., `no_load_current`).
+
+### Stiction
+
+We add a dither to each joint when it is moving below a specified speed (i.e., `dither_speed`) to avoid stiction. It periodically pushes the joint in alternating directions so that the joint is moving for the majority of time.
+The dithering frequency is half of the publishing rate of the `/<namespace>/joint_states` topic.
+The dithering magnitude dither matches that of the static friction torque, given by
+
+```math
+\tau_s = \mu_s\tau
+```
+
+where $\mu_s$ is the coefficient of static friction associated with the torque (i.e., `static_friction_coefficient`).
+
+**WARNING: excessive dithering WILL cause heat and wear on the joints**
```

Comment:
GitHub has provisions for callouts like this in their markdown syntax: https://github.com/orgs/community/discussions/16925